diff -ur tk8.5.9.orig/win/tcl.m4 tk8.5.9/win/tcl.m4
--- tk8.5.9.orig/win/tcl.m4	2010-09-02 22:12:14 +0400
+++ tk8.5.9/win/tcl.m4	2011-03-24 23:53:08 +0300
@@ -11,10 +11,10 @@
 # Results:
 #
 #	Adds the following arguments to configure:
-#		--with-tcl=...
+#		--with-tcl-prefix=...
 #
 #	Sets the following vars:
-#		TCL_BIN_DIR	Full path to the tclConfig.sh file
+#		TCL_DIR	Full path to the tclConfig.sh file
 #------------------------------------------------------------------------
 
 AC_DEFUN([SC_PATH_TCLCONFIG], [
@@ -28,18 +28,18 @@
 	TCL_BIN_DIR_DEFAULT=../../tcl/win
     fi
 
-    AC_ARG_WITH(tcl, [  --with-tcl=DIR          use Tcl 8.5 binaries from DIR],
-	    TCL_BIN_DIR=$withval, TCL_BIN_DIR=`cd $TCL_BIN_DIR_DEFAULT; pwd`)
-    if test ! -d $TCL_BIN_DIR; then
-	AC_MSG_ERROR(Tcl directory $TCL_BIN_DIR does not exist)
+    AC_ARG_WITH(tcl-prefix, [  --with-tcl-prefix=DIR          use Tcl 8.5 binaries from DIR/bin],
+	    TCL_DIR=$withval, TCL_DIR=`cd $TCL_BIN_DIR_DEFAULT; pwd`)
+    if test ! -d $TCL_DIR; then
+	AC_MSG_ERROR(Tcl directory $TCL_DIR does not exist)
     fi
-    if test ! -f $TCL_BIN_DIR/tclConfig.sh; then
-	if test ! -f $TCL_BIN_DIR/../unix/tclConfig.sh; then
-	    AC_MSG_ERROR(There is no tclConfig.sh in $TCL_BIN_DIR:  perhaps you did not specify the Tcl *build* directory (not the toplevel Tcl directory) or you forgot to configure Tcl?)
-	fi
-	TCL_BIN_DIR=`cd ${TCL_BIN_DIR}/../unix; pwd`
+    if test ! -f $TCL_DIR/bin/tclConfig.sh; then
+	if test ! -f $TCL_DIR/../unix/tclConfig.sh; then
+	  AC_MSG_ERROR(There is no tclConfig.sh in $TCL_DIR:  perhaps you did not specify the Tcl *build* directory (not the toplevel Tcl directory) or you forgot to configure Tcl?)
+	fi
+	TCL_DIR=`cd ${TCL_DIR}/../unix; pwd`
     fi
-    AC_MSG_RESULT($TCL_BIN_DIR/tclConfig.sh)
+    AC_MSG_RESULT($TCL_DIR/bin/tclConfig.sh)
 ])
 
 #------------------------------------------------------------------------
@@ -68,16 +68,16 @@
 	TK_BIN_DIR_DEFAULT=../../tk/win
     fi
 
-    AC_ARG_WITH(tk, [  --with-tk=DIR          use Tk 8.5 binaries from DIR],
-	    TK_BIN_DIR=$withval, TK_BIN_DIR=`cd $TK_BIN_DIR_DEFAULT; pwd`)
-    if test ! -d $TK_BIN_DIR; then
-	AC_MSG_ERROR(Tk directory $TK_BIN_DIR does not exist)
+    AC_ARG_WITH(tk-prefix, [  --with-tk-prefix=DIR          use Tk 8.5 binaries from DIR/bin],
+	    TK_DIR=$withval, TK_DIR=`cd $TK_BIN_DIR_DEFAULT; pwd`)
+    if test ! -d $TK_DIR; then
+	AC_MSG_ERROR(Tk directory $TK_DIR does not exist)
     fi
-    if test ! -f $TK_BIN_DIR/tkConfig.sh; then
-	AC_MSG_ERROR(There is no tkConfig.sh in $TK_BIN_DIR:  perhaps you did not specify the Tk *build* directory (not the toplevel Tk directory) or you forgot to configure Tk?)
+    if test ! -f $TK_DIR/bin/tkConfig.sh; then
+	AC_MSG_ERROR(There is no tkConfig.sh in $TK_DIR:  perhaps you did not specify the Tk *build* directory (not the toplevel Tk directory) or you forgot to configure Tk?)
     fi
 
-    AC_MSG_RESULT([$TK_BIN_DIR/tkConfig.sh])
+    AC_MSG_RESULT([$TK_DIR/bin/tkConfig.sh])
 ])
 
 #------------------------------------------------------------------------
@@ -88,43 +88,28 @@
 # Arguments:
 #
 #	Requires the following vars to be set:
-#		TCL_BIN_DIR
+#		TCL_DIR
 #
 # Results:
 #
 #	Subst the following vars:
-#		TCL_BIN_DIR
+#		TCL_DIR
 #		TCL_SRC_DIR
 #		TCL_LIB_FILE
 #
 #------------------------------------------------------------------------
 
 AC_DEFUN([SC_LOAD_TCLCONFIG], [
-    AC_MSG_CHECKING([for existence of $TCL_BIN_DIR/tclConfig.sh])
+    AC_MSG_CHECKING([for existence of $TCL_DIR/bin/tclConfig.sh])
 
-    if test -f "$TCL_BIN_DIR/tclConfig.sh" ; then
+    if test -f "$TCL_DIR/bin/tclConfig.sh" ; then
         AC_MSG_RESULT([loading])
-	. $TCL_BIN_DIR/tclConfig.sh
+	. $TCL_DIR/bin/tclConfig.sh
     else
         AC_MSG_RESULT([file not found])
     fi
 
     #
-    # If the TCL_BIN_DIR is the build directory (not the install directory),
-    # then set the common variable name to the value of the build variables.
-    # For example, the variable TCL_LIB_SPEC will be set to the value
-    # of TCL_BUILD_LIB_SPEC. An extension should make use of TCL_LIB_SPEC
-    # instead of TCL_BUILD_LIB_SPEC since it will work with both an
-    # installed and uninstalled version of Tcl.
-    #
-
-    if test -f $TCL_BIN_DIR/Makefile ; then
-        TCL_LIB_SPEC=${TCL_BUILD_LIB_SPEC}
-        TCL_STUB_LIB_SPEC=${TCL_BUILD_STUB_LIB_SPEC}
-        TCL_STUB_LIB_PATH=${TCL_BUILD_STUB_LIB_PATH}
-    fi
-
-    #
     # eval is required to do the TCL_DBGX substitution
     #
 
@@ -137,7 +122,7 @@
     eval "TCL_STUB_LIB_SPEC=\"${TCL_STUB_LIB_SPEC}\""
 
     AC_SUBST(TCL_VERSION)
-    AC_SUBST(TCL_BIN_DIR)
+    AC_SUBST(TCL_DIR)
     AC_SUBST(TCL_SRC_DIR)
 
     AC_SUBST(TCL_LIB_FILE)
@@ -799,27 +784,27 @@
 #		--with-tcl=...
 #
 #	Defines the following vars:
-#		TCL_BIN_DIR	Full path to the tcl build dir.
+#		TCL_DIR	Full path to the tcl build dir.
 #------------------------------------------------------------------------
 
-AC_DEFUN([SC_WITH_TCL], [
+AC_DEFUN([SC_WITH_TCL_PREFIX], [
     if test -d ../../tcl8.5$1/win;  then
-	TCL_BIN_DEFAULT=../../tcl8.5$1/win
+	TCL_PREF_DEFAULT=../../tcl8.5$1/win
     else
-	TCL_BIN_DEFAULT=../../tcl8.5/win
+	TCL_PREF_DEFAULT=../../tcl8.5/win
     fi
 
-    AC_ARG_WITH(tcl, [  --with-tcl=DIR          use Tcl 8.5 binaries from DIR],
-	    TCL_BIN_DIR=$withval, TCL_BIN_DIR=`cd $TCL_BIN_DEFAULT; pwd`)
-    if test ! -d $TCL_BIN_DIR; then
-	AC_MSG_ERROR(Tcl directory $TCL_BIN_DIR does not exist)
+    AC_ARG_WITH(tcl, [  --with-tcl-prefix=DIR          use Tcl 8.5 from DIR/bin],
+	    TCL_DIR=$withval, TCL_DIR=`cd $TCL_PREF_DEFAULT; pwd`)
+    if test ! -d $TCL_DIR; then
+	AC_MSG_ERROR(Tcl directory $TCL_DIR does not exist)
     fi
-    if test ! -f $TCL_BIN_DIR/Makefile; then
-	AC_MSG_ERROR(There is no Makefile in $TCL_BIN_DIR:  perhaps you did not specify the Tcl *build* directory (not the toplevel Tcl directory) or you forgot to configure Tcl?)
+    if test ! -f $TCL_DIR/Makefile; then
+	AC_MSG_ERROR(There is no Makefile in $TCL_DIR:  perhaps you did not specify the Tcl *build* directory (not the toplevel Tcl directory) or you forgot to configure Tcl?)
     else
-	echo "building against Tcl binaries in: $TCL_BIN_DIR"
+	echo "building against Tcl binaries in: $TCL_DIR"
     fi
-    AC_SUBST(TCL_BIN_DIR)
+    AC_SUBST(TCL_DIR)
 ])
 
 #------------------------------------------------------------------------
@@ -890,7 +875,7 @@
 
 AC_DEFUN([SC_BUILD_TCLSH], [
     AC_MSG_CHECKING([for tclsh in Tcl build directory])
-    BUILD_TCLSH=${TCL_BIN_DIR}/tclsh${TCL_MAJOR_VERSION}${TCL_MINOR_VERSION}${TCL_DBGX}${EXEEXT}
+    BUILD_TCLSH=${TCL_DIR}/bin/tclsh${TCL_MAJOR_VERSION}${TCL_MINOR_VERSION}${TCL_DBGX}${EXEEXT}
     AC_MSG_RESULT($BUILD_TCLSH)
     AC_SUBST(BUILD_TCLSH)
 ])
diff -ur tk8.5.9.orig/win/Makefile.in tk8.5.9/win/Makefile.in
--- tk8.5.9.orig/win/Makefile.in	2010-09-02 22:12:14 +0400
+++ tk8.5.9/win/Makefile.in	2011-03-25 00:10:19 +0300
@@ -89,7 +89,7 @@
 
 # The directory containing the Tcl library archive file appropriate
 # for this version of Tk:
-TCL_BIN_DIR		= @TCL_BIN_DIR@
+TCL_DIR		= @TCL_DIR@
 
 # The directory containing the Tcl sources and headers appropriate
 # for this version of Tk ("srcdir" will be replaced or has already
@@ -106,8 +106,8 @@
 CYGPATH			= @CYGPATH@
 
 # The name of the Tcl library.
-TCL_LIB_FILE	  = "$(shell $(CYGPATH) '@TCL_BIN_DIR@/@TCL_LIB_FILE@')"
-TCL_STUB_LIB_FILE = "$(shell $(CYGPATH) '@TCL_BIN_DIR@/@TCL_STUB_LIB_FILE@')"
+TCL_LIB_FILE	  = "$(shell $(CYGPATH) '@TCL_DIR@/lib/@TCL_LIB_FILE@')"
+TCL_STUB_LIB_FILE = "$(shell $(CYGPATH) '@TCL_DIR@/lib/@TCL_STUB_LIB_FILE@')"
 
 SRC_DIR			= @srcdir@
 ROOT_DIR		= $(SRC_DIR)/..
@@ -399,7 +399,7 @@
 SHELL_ENV = \
 	@TCL_LIBRARY="$(TCL_SRC_DIR_NATIVE)/library"; export TCL_LIBRARY; \
 	TK_LIBRARY="$(ROOT_DIR_NATIVE)/library"; export TK_LIBRARY; \
-	PATH="$(TCL_BIN_DIR):$(PATH)"; export PATH;
+	PATH="$(TCL_DIR)/bin:$(PATH)"; export PATH;
 
 # Main targets.  The default target -- all -- builds the binaries, 
 # performs any post processing on libraries or documents.
@@ -417,7 +417,7 @@
 
 winhelp: $(TCL_SRC_DIR)/tools/man2help.tcl $(MAN2TCL)
 	$(TCL_EXE) "$(TCL_SRC_DIR_NATIVE)/tools/man2help.tcl" tcl "$(VER)" $(CORE_DOCS)
-	$(COPY) "$(TCL_BIN_DIR)/tcl.hpj" ./
+	$(COPY) "$(TCL_DIR)/bin/tcl.hpj" ./
 	hcw /c /e tcl.hpj
 	$(COPY) ./tcl$(VER).cnt ./TCL$(VER).HLP "$(TCL_SRC_DIR_NATIVE)"/tools/
 
@@ -453,7 +453,7 @@
 gdb: binaries
 	@echo "set env TCL_LIBRARY=$(TCL_SRC_DIR_NATIVE)/library" > gdb.run
 	@echo "set env TK_LIBRARY=$(ROOT_DIR_NATIVE)/library" >> gdb.run
-	PATH="$(TCL_BIN_DIR):$(PATH)"; export PATH; \
+	PATH="$(TCL_DIR)/bin:$(PATH)"; export PATH; \
 	gdb ./$(WISH) --command=gdb.run
 	@$(RM) gdb.run
 
diff -ur tk8.5.9.orig/win/configure.in tk8.5.9/win/configure.in
--- tk8.5.9.orig/win/configure.in	2010-09-02 22:12:14 +0400
+++ tk8.5.9/win/configure.in	2011-03-25 00:07:35 +0300
@@ -268,7 +268,7 @@
 AC_SUBST(TCL_MINOR_VERSION)
 AC_SUBST(TCL_PATCH_LEVEL)
 AC_SUBST(TCL_SRC_DIR)
-AC_SUBST(TCL_BIN_DIR)
+AC_SUBST(TCL_DIR)
 AC_SUBST(TCL_DBGX)
 AC_SUBST(CFG_TK_SHARED_LIB_SUFFIX)
 AC_SUBST(CFG_TK_UNSHARED_LIB_SUFFIX)
